### PR TITLE
Adding a README

### PR DESCRIPTION
**Pull request by kv-sc**
This commit adds basic README to make start of development easier

### CODE REVIEW

File: llvm/tools/llvm-snippy/README.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+```
+set(CMAKE_BUILD_TYPE RELEASE CACHE STRING "")
+set(CMAKE_INSTALL_PREFIX "/path/to/install" CACHE STRING "")
+set(CMAKE_C_COMPILER clang CACHE STRING "")
+set(CMAKE_CXX_COMPILER clang++ CACHE STRING "")
+set(LLVM_ENABLE_LLD ON CACHE BOOL "")
+set(LLVM_ENABLE_PROJECTS
+  lld
+  clang
+  CACHE STRING "")
```
Comment (asi-sc) [inline llvm/tools/llvm-snippy/README.md:24]: I don't think these projects are necessary for building snippy.

File: llvm/tools/llvm-snippy/README.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+Let's briefly explain the content of the layout.yaml file:
+
+### options
+
+This section allows you to set various command-line options. They can also be set directly in the command line. Here's a breakdown of some options:
+
+- march: Target architecture (e.g., "riscv64-linux-gnu").
+- model-plugin: Model plugin to use (e.g., "None").
+- num-instrs: Number of instructions.
```
Comment (asi-sc) [inline llvm/tools/llvm-snippy/README.md:118]: ... to generate

File: llvm/tools/llvm-snippy/README.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+- o: Base name for the output file.
+- init-regs-in-elf: Initialize non-fixed registers in the ELF.
```
Comment (asi-sc) [inline llvm/tools/llvm-snippy/README.md:120]: available registers?

File: llvm/tools/llvm-snippy/README.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+```
+
+The llvm-snippy generator also supports various models through a unified interface. We plan to open-source SAIL-based and spike-based models that are already operational within Syntacore. Currently, the primary mode is without a model unless you write and integrate your own.
+
+```
+num-instrs: 10000
+o: "snippet.elf"
+```
+
+These options specify the number of instructions and the base name of the output file. You can set num-instrs to "all" and the entire section will be filled.
```
Comment (asi-sc) [inline llvm/tools/llvm-snippy/README.md:139]: RX section

File: llvm/tools/llvm-snippy/README.md [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+```
+init-regs-in-elf: true
+```
+
+It initializes all unfixed registers with random values at the beginning of the snippet. This option helps avoid non-determinism due to different register settings before a function call.
```
Comment (asi-sc) [inline llvm/tools/llvm-snippy/README.md:145]: unfixed? Are you sure that it's the best term? Personally I don't understand it :). I suggest using available, reserved and so on.

File: llvm/tools/llvm-snippy/README.md [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+
+### sections
+
+Sections are used to either store code (rx sections) or read and write data (r and rw sections). Full support for rwx sections has not yet been open-sourced.
+
+The section names are arbitrary and conditional—they will be mangled in the final snippet. However, they are useful for understanding the structure.
+
+In the current open-source implementation, there is no distinction between VMA and LMA. The latter is ignored for practical purposes, but it is recommended to specify them as identical for now, just in case.
+
+### Раздел histogram
```
Comment (asi-sc) [inline llvm/tools/llvm-snippy/README.md:155]: Razdel

File: llvm/tools/llvm-snippy/README.md [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+
+Weights are relative. A weight of 10.0 means that the primary instruction will occur approximately 10 times more often than an instruction marked with weight 1.0. All weights are summed, and to determine the probability of an instruction, the weight should be divided by the sum of weights.
+
+To see possible instruction names for the histogram, you can use the following command:
+
+```
+./llvm-snippy -march=riscv64-linux-gnu -list-opcode-names
+```
+
+At the time of open-sourcing, llvm-snippy works with all RISCV instructions.
```
Comment (asi-sc) [inline llvm/tools/llvm-snippy/README.md:167]: available in the LLVM backend

File: llvm/tools/llvm-snippy/README.md [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+}
+```
+
+You can reproduce the same generation by repeating the call with the same seed.
+
+```
+> ${SNIPPY_INSTALL}/llvm-snippy --seed=1703685916769088352 layout.yaml
+```
+
+Of course, not everything is covered in this quick start. We will open-source detailed documentation soon. For now, the help command is available.
```
Comment (asi-sc) [inline llvm/tools/llvm-snippy/README.md:246]: open-source -> publish?

File: llvm/tools/llvm-snippy/README.md (hunk 1 above)
Comment (kv-sc) [reply llvm/tools/llvm-snippy/README.md]: Sure. Also removed X86

File: llvm/tools/llvm-snippy/README.md (hunk 2 above)
Comment (kv-sc) [reply llvm/tools/llvm-snippy/README.md]: Done

File: llvm/tools/llvm-snippy/README.md (hunk 3 above)
Comment (kv-sc) [reply llvm/tools/llvm-snippy/README.md]: Yep

File: llvm/tools/llvm-snippy/README.md (hunk 4 above)
Comment (kv-sc) [reply llvm/tools/llvm-snippy/README.md]: code section

File: llvm/tools/llvm-snippy/README.md (hunk 5 above)
Comment (kv-sc) [reply llvm/tools/llvm-snippy/README.md]: non-reserved

File: llvm/tools/llvm-snippy/README.md (hunk 6 above)
Comment (kv-sc) [reply llvm/tools/llvm-snippy/README.md]: Fixed

File: llvm/tools/llvm-snippy/README.md (hunk 7 above)
Comment (kv-sc) [reply llvm/tools/llvm-snippy/README.md]: Good point!

File: llvm/tools/llvm-snippy/README.md (hunk 8 above)
Comment (kv-sc) [reply llvm/tools/llvm-snippy/README.md]: Yep.

**Review by asi-sc — APPROVED** [review]

**Pull request merged**